src/StoreApi/Utilities/ProductQueryFilters.php

```diff
@@ -211,4 +211,91 @@ public function get_rating_counts( $request ) {
 
 		return array_map( 'absint', wp_list_pluck( $results, 'product_count', 'rounded_average_rating' ) );
 	}
+
+	/**
+	 * Gets product by filtered terms.
+	 *
+	 * @since TBD
+	 * @param string $taxonomy Taxonomy name.
+	 * @param array  $term_ids Term IDs.
+	 * @param string $query_type all | any.
+	 * @return array
+	 */
+	public function get_product_by_filtered_terms( $taxonomy = '', $term_ids = array(), $query_type = 'any' ) {
+		global $wpdb;
+
+		$transient_key  = 'wc_get_product_by_filtered_terms_' . md5(
```

Comment:
What do you think about following the same pattern we discussed on https://github.com/woocommerce/woocommerce-blocks/pull/8645#issuecomment-1466549055 and only cache the macro query instead?